Doc/using/windows.rst

```diff
@@ -602,6 +602,49 @@ existed)::
 
     C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\Python 3.9
 
+.. _win-utf8-mode:
+
+Using UTF-8 mode
+================
+
+.. versionadded:: 3.7
+
+Windows doesn't use UTF-8 for the system encoding (the ANSI Code Page).
+And Python uses the system encoding for the default encoding of text files.
+
+It may cause trouble because UTF-8 is very widely used on the internet
+and most Unix systems, including WSL (Windows Subsystem for Linux).
+
+You can use the UTF-8 mode to change the default text encoding to UTF-8.
+You can enable the UTF-8 mode via ``-X utf8`` command option, or
+``PYTHONUTF8=1`` environement variable.  See :envvar:`PYTHONUTF8` for
+the UTF-8 mode, and :ref:`setting-envvars` for how to modify
+environment variables.
+
+When UTF-8 mode is enabled:
+
+* :func:`locale.getpreferredencoding` returns ``'UTF-8'`` instead of
+  the legacy system encoding.  This function is used for the default
+  text encoding in many places, including :func:`open`, :class:`Popen`,
+  :meth:`Path.read_text`, etc.
+* :data:`sys.stdin`, :data:`sys.stdout`, and :data:`sys.stderr`
+  all use UTF-8 as their text encoding.
+
+.. note::
+   Even when the UTF-8 mode is disabled, Python uses UTF-8 on Windows for:
+
+   * :data:`sys.stdin`, :data:`sys.stdout`, and :data:`sys.stderr`
+     when they are console (see :pep:`528` for detail).
+   * The filesystem encoding (see :pep:`529` for detail).
+
+
+You can use the "mbcs" codec for the system encoding in UTF-8 mode.
+
+And you can use the legacy encoding for filesystem encoding via
+:func:`sys._enablelegacywindowsfsencoding` or
+:envvar:`PYTHONLEGACYWINDOWSFSENCODING` in UTF-8 mode too.
```

Comment:
I still wish it was decided for UTF-8 mode to take precedence over `PYTHONLEGACYWINDOWSFSENCODING` (but not `sys._enablelegacywindowsfsencoding`). The legacy filesystem encoding configuration modifies the default behavior, but UTF-8 mode explicitly overrides default behavior. For example, if we had UTF-8 mode in Python 3.5, before the default filesystem encoding changed to UTF-8 in Windows, I think it would have been justified for UTF-8 mode to override the default filesystem encoding and bypass the Windows MBS API (e.g. `open`, `CreateFileA`) to essentially work the way the UTF-8 filesystem encoding works now, i.e. manually decode and call the WCS API function. Whether someone would have been ambitious enough to do that before Steve Dower got around to it in 3.6 is another matter (probably no).

Reply:
I want to backport this pull request to Python 3.7 and 3.8.
So I am documenting the current behavior.

Please discuss about changing behavior in python-dev or disuss.python.org.

Reply:
I removed this paragraph because most users don't care about legacy fs encoding.